### PR TITLE
530 get cdc searchapi visitors information

### PR DESCRIPTION
**Pull request by kpapag**

### CODE REVIEW

**Review by john-shepherdson — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
     "lodash": "4.17.21",
     "method-override": "3.0.0",
     "node-fetch": "2.6.9",
+    "node-ipinfo": "^3.4.0",
```
Comment (matthew-morris-cessda) [inline package.json:33]: Could this be replaced by a fixed version, i.e. `"node-ipinfo": "3.4.0",`?

File: package-lock.json
Comment (matthew-morris-cessda) [inline package-lock.json:1]: The `package-lock.json` has been reverted to the `v1` format, which is caused by old versions of NPM.

Could you update your version of Node.js and NPM, then run `npm install` again.

File: server/helper.ts [hunk 2]
```diff
@@ -240,6 +241,25 @@ function externalApiV2() {
       return;
     }
 
+    //Get Visitors Information For Prom Metrics
+    const ip: string = req.headers['x-forwarded-for'] as string || req.socket.remoteAddress as string;
```
Comment (matthew-morris-cessda) [inline server/helper.ts:245]: `req.headers['x-forwarded-for'] as string` should be `req.headers['x-forwarded-for'] as string | undefined`

Please don't use type assertions like `as` unless you can verify the type.

File: server/helper.ts [hunk 3]
```diff
@@ -240,6 +241,25 @@ function externalApiV2() {
...
+    const ipinfoWrapper = new IPinfoWrapper(""); //token must be parsed here
+    //timeouts defaults to 5000 i.e. 5 seconds - can be changed if needed
+    ipinfoWrapper.lookupIp(ip).then((response: IPinfo) => {
+      searchAPIClientIPGauge.labels({ searchAPIClientIP: response.ip}).inc();
+      searchAPIClientCountryGauge.labels({ searchAPIClientCountry: response.country}).inc();
+    })
+    .catch((error) => {
+      console.log(error);
```
Comment (matthew-morris-cessda) [inline server/helper.ts:253]: This is redundant

**Review by matthew-morris-cessda — CHANGES_REQUESTED** [review]

**Review by matthew-morris-cessda — APPROVED** [review]
Looks good

**Pull request merged**